# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/Device.h
# Commit: 843c7d19787b
# Full Hash: 843c7d19787b5beb9caa2fbf71192b174738bb92
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2023-10-17 03:58:35
# Regressor Bug: 1838693
# File Overlap Count: 1
# Description:
#   Bug 1838693 Part 1: Stub in 'lose the device' and trigger it on device lost errors. r=webgpu-reviewers,nical
#   
#   This creates a WebGPUParent::LoseDevice entry point, which informs the
#   WebGPUChild that the device has been lost. The child side is largely a
#   stub, as a later part will rationalize the handling of the device.lost
# ==============================================================================

diff -r e0dd0b10e8fd -r 843c7d19787b dom/webgpu/Device.h
--- a/dom/webgpu/Device.h	Mon Oct 16 15:29:21 2023 +0000
+++ b/dom/webgpu/Device.h	Mon Oct 16 15:31:59 2023 +0000
@@ -45,6 +45,7 @@
 template <typename T>
 class Sequence;
 class GPUBufferOrGPUTexture;
+enum class GPUDeviceLostReason : uint8_t;
 enum class GPUErrorFilter : uint8_t;
 enum class GPUFeatureName : uint8_t;
 class GPULogCallback;
@@ -121,6 +122,8 @@
   void SetLabel(const nsAString& aLabel);
   dom::Promise* GetLost(ErrorResult& aRv);
   dom::Promise* MaybeGetLost() const { return mLostPromise; }
+  void ResolveLost(Maybe<dom::GPUDeviceLostReason> aReason,
+                   const nsAString& aMessage);
 
   const RefPtr<SupportedFeatures>& Features() const { return mFeatures; }
   const RefPtr<SupportedLimits>& Limits() const { return mLimits; }